service/frontend/fx.go

```diff
@@ -275,26 +319,118 @@ func VisibilityManagerProvider(
-		namespaceReplicationQueue = serviceResource.GetNamespaceReplicationQueue()
+		replicatorNamespaceReplicationQueue = namespaceReplicationQueue
 	}
-	return namespaceReplicationQueue
+	return replicatorNamespaceReplicationQueue
+}
+
+func ServiceResolverProvider(membershipMonitor membership.Monitor) (membership.ServiceResolver, error) {
+	return membershipMonitor.GetResolver(common.FrontendServiceName)
+}
+
+func AdminHandlerProvider(
+	params *resource.BootstrapParams,
+	config *Config,
+	replicatorNamespaceReplicationQueue FEReplicatorNamespaceReplicationQueue,
+	esConfig *esclient.Config,
+	esClient esclient.Client,
+	visibilityMrg manager.VisibilityManager,
+	logger resource.SnTaggedLogger,
+	persistenceExecutionManager persistence.ExecutionManager,
+	namespaceReplicationQueue persistence.NamespaceReplicationQueue,
+	taskManager persistence.TaskManager,
+	clusterMetadataManager persistence.ClusterMetadataManager,
+	persistenceMetadataManager persistence.MetadataManager,
+	clientFactory client.Factory,
+	clientBean client.Bean,
+	historyClient historyservice.HistoryServiceClient,
+	sdkClient sdkclient.Client,
+	membershipMonitor membership.Monitor,
+	archiverProvider provider.ArchiverProvider,
+	metricsClient metrics.Client,
+	namespaceRegistry namespace.Registry,
+	saProvider searchattribute.Provider,
+	saManager searchattribute.Manager,
+	clusterMetadata cluster.Metadata,
+	archivalMetadata archiver.ArchivalMetadata,
+) *AdminHandler {
+	return NewAdminHandler(
+		params,
+		config,
+		namespaceReplicationQueue,
+		replicatorNamespaceReplicationQueue,
+		esConfig,
+		esClient,
+		visibilityMrg,
+		logger,
+		persistenceExecutionManager,
+		taskManager,
+		clusterMetadataManager,
+		persistenceMetadataManager,
+		clientFactory,
+		clientBean,
+		historyClient,
+		sdkClient,
+		membershipMonitor,
+		archiverProvider,
+		metricsClient,
+		namespaceRegistry,
+		saProvider,
+		saManager,
+		clusterMetadata,
+		archivalMetadata,
+	)
 }
 
 func HandlerProvider(
 	params *resource.BootstrapParams,
-	serviceResource resource.Resource,
 	serviceConfig *Config,
-	namespaceReplicationQueue persistence.NamespaceReplicationQueue,
+	namespaceReplicationQueue FEReplicatorNamespaceReplicationQueue,
 	visibilityMgr manager.VisibilityManager,
+	logger resource.SnTaggedLogger,
+	throttledLogger resource.ThrottledLogger,
+	persistenceExecutionManager persistence.ExecutionManager,
+	clusterMetadataManager persistence.ClusterMetadataManager,
+	persistenceMetadataManager persistence.MetadataManager,
+	clientBean client.Bean,
+	historyClient historyservice.HistoryServiceClient,
+	matchingClient resource.MatchingClient,
+	archiverProvider provider.ArchiverProvider,
+	metricsClient metrics.Client,
+	payloadSerializer serialization.Serializer,
+	timeSource clock.TimeSource,
+	namespaceRegistry namespace.Registry,
+	saMapper searchattribute.Mapper,
+	saProvider searchattribute.Provider,
+	clusterMetadata cluster.Metadata,
+	archivalMetadata archiver.ArchivalMetadata,
```

Comment:
I would be against it because most of objects do require distinct sets of arguments on one hand and this list is big due to tech debt inside relevant instatiated classes. This would make sense to do in parts of code that are not custom providers. Additionally, we want to avoid creating "context/resource/etc" classes that might become used to plumb objects through the system.